### ISSUE TITLE
Snapshot restore (SQL server)

### ISSUE THREAD

**Issue by MihailsKuzmins** [Zaid-Ajaj/ThrowawayDb#9]
It would be great to provide some snapshot-restore functionality. For an SQL server the basic SQL scripts are
```sql
-- Create a snapshot
CREATE DATABASE MySnapshot ON  
(
    NAME = MyDb, FILENAME = 'C:\Program Files\Microsoft SQL Server\MSSQL15.SQLEXPRESS\MSSQL\DATA\MySnapshot.ss' )  
AS SNAPSHOT OF MyDb;  
GO

-- Restore a snapshot
RESTORE DATABASE MyDb
FROM DATABASE_SNAPSHOT = 'MySnapshot';
GO

-- Cleanup
DROP database MySnapshot
DROP database MyDb
```
Would it be possible for you to try to implement this feature or should I try to create a PR for it?

**Comment by Zaid-Ajaj** [comment]
Hi @MihailsKuzmins, that would be a great addition. I haven't played with snapshots before so I am not sure how to implement it, especially not sure where that `it_gxie_72be4a7284` database is coming from.  

It would be awesome if you could try and create a PR for it 🙏 let me know if you need anything :smile:

**Comment by MihailsKuzmins** [comment]
Hello, @Zaid-Ajaj, sorry, my bad 😄. I forgot to change it's name, of course it should be `MyDb`, corrected my comment.
So I will try to produce some results by the end of this week ;)

**Comment by Zaid-Ajaj** [comment]
No worries @MihailsKuzmins that sounds great! I look forward to the PR but of course, no rush at all :smile:

**Comment by MihailsKuzmins** [comment]
Hello @Zaid-Ajaj, as promised here I am with my PR #10. I was able to test the code in my database with several simultaneous test cases (each case includes performing some operations and restoring a snapshot created during the initialisation). So I hope the code I've written is OK. Please take a look at it and provide your feedback :)
Thank you in advance ;)

**Comment by Zaid-Ajaj** [comment]
Thanks again @MihailsKuzmins, a package has been published for ThrowawayDb as of v1.3 via  #10

**Comment by MihailsKuzmins** [comment]
> Thanks again @MihailsKuzmins, a package has been published for ThrowawayDb as of v1.3 via #10

Thank you very much! 🙏 Looking forward to using the new version from NuGet
